var/spack/repos/builtin/packages/py-flake8/package.py

```diff
@@ -57,11 +57,12 @@ class PyFlake8(Package):
     # mccabe >= 0.2.1, < 0.5
     depends_on('py-mccabe@0.2.1:0.4.0', when='@2.5.4', type=('build', 'run'))
 
-    depends_on('py-configparser', when='^python@:3.3.999', type=('build', 'run'))
-    depends_on('py-enum34', when='^python@:3.1.999', type=('build', 'run'))
+    # These dependencies breaks concretization
+    # See https://github.com/LLNL/spack/issues/2793
+    # depends_on('py-configparser', when='^python@:3.3.999', type=('build', 'run'))  # noqa
+    # depends_on('py-enum34', when='^python@:3.1.999', type=('build', 'run'))
```

Comment:
I believe you can use `python@3.3` here; `python@3.3` acts like a set that also matches `python@3.3.x`.  Not sure where else it happen in this PR (and this one's in a comment); but something to look out for.

Are you able to grep through and fix for places where `.999` occurs?